pkg/intercp/client/pool.go

```diff
@@ -90,19 +90,19 @@ func (c *Pool) SetTLSConfig(tlsCfg *TLSConfig) {
 func (c *Pool) StartCleanup(ctx context.Context, ticker *time.Ticker) {
 	for {
 		select {
-		case <-ticker.C:
-			c.cleanup()
+		case now := <-ticker.C:
```

Comment:
Why do we need this change? Isn't it strange that we compare times from different sources, because `lastAccessTime` gets the time from `c.now()` and `ticker` probably gets the time from the runtime `now()`

Reply:
The advantage of this is that you can simulate time passing more easily (just tick with different times). I think if you use a different now that's not related to your ticker you'd have other problems anyway regardless of which time you use

Reply:
But this gives us not a great `Pool` interface. We have a hidden assumption that `now` in:
```go
func NewPool(..., now func() time.Time) *Pool
```
should be related to `ticker`:
```go
func (c *Pool) StartCleanup(..., ticker *time.Ticker)
```

Reply:
But if you don't tick accurately your pool is going to be broken right? I could put the ticker in `NewPool` if you prefer

Reply:
It would be nice if we could parametrize the ticker with `now()` function, but it's not possible, so I guess the existing approach is fine.